packages/stencil-library/src/components/radio-group/radio-group.e2e.ts

```diff
@@ -0,0 +1,528 @@
+            <njwds-radio value="douglass">Frederick Douglass</njwds-radio>
+      </njwds-radio-group>`);
+      expect(await radioGroup.getProperty('value')).toBe('douglass');
+      const truthInput = await radioGroup.find('input[value="truth"]');
+      await truthInput.click();
+      expect(await radioGroup.getProperty('value')).toBe('truth');
+    });
+
+    it('when the value prop is programmatically set, selects the correct njwds-radio element', async () => {
+      const { radioGroup, page } = await renderAndGetRadioGroupAndPage(`<njwds-radio-group
+          name="historical-figures"
+        >
+            <njwds-radio value="truth">Sojourner Truth</njwds-radio>
+            <njwds-radio value="douglass">Frederick Douglass</njwds-radio>
+      </njwds-radio-group>`);
+      const douglassInput = await radioGroup.find('input[value="douglass"]');
+      expect(await douglassInput.getProperty('checked')).toBe(false);
+
+      radioGroup.setAttribute('value', 'douglass');
+      await page.waitForChanges();
+      expect(await douglassInput.getProperty('checked')).toBe(true);
+    });
+  });
+
+  describe('getValidity() method', () => {
+    it('returns a value missing validity state when required input is missing', async () => {
+      const { radioGroup } = await renderAndGetRadioGroupAndPage(`<njwds-radio-group
+          name="historical-figures"
+          required
+        >
+            <njwds-radio value="truth">Sojourner Truth</njwds-radio>
+            <njwds-radio value="douglass">Frederick Douglass</njwds-radio>
+      </njwds-radio-group>`);
+
+      const validity: RadioGroupValidityState = await radioGroup.callMethod('getValidity');
+      expect(validity.valid).toBe(false);
+      expect(validity.valueMissing).toBe(true);
+    });
+
+    it('returns a valid validity state when input is not required', async () => {
+      const { radioGroup } = await renderAndGetRadioGroupAndPage(`<njwds-radio-group
+          name="historical-figures"
+        >
+            <njwds-radio value="truth">Sojourner Truth</njwds-radio>
+            <njwds-radio value="douglass">Frederick Douglass</njwds-radio>
+      </njwds-radio-group>`);
+
+      const validity: RadioGroupValidityState = await radioGroup.callMethod('getValidity');
+      expect(validity.valid).toBe(true);
+      expect(validity.valueMissing).toBe(false);
+    });
+  });
+
+  it('sets the validationMessage prop to a non-empty string by default', async () => {
+    const { radioGroup } = await renderAndGetRadioGroupAndPage(`<njwds-radio-group
+        name="historical-figures"
+      >
+          <njwds-radio value="truth">Sojourner Truth</njwds-radio>
+          <njwds-radio value="douglass">Frederick Douglass</njwds-radio>
+    </njwds-radio-group>`);
+    const validationMessage = await radioGroup.getProperty('validationMessage');
+    expect(typeof validationMessage).toBe('string');
+    expect(validationMessage.length).toBeGreaterThan(0);
```

Comment:
Revised to test the specific default validation message! I kept it more generic because I wanted to keep the test robust against changes to the specific wording of the default message — but thinking about it again, the tradeoff of having to change one test is worth making sure the correct message is displayed.